<launch>
	<!--
	<env name = "GSCAM_CONFIG" 
		value = "v4l2src device=/dev/video0 ! video/x-raw-rgb,framerate=25/1 ! ffmpegcolorspace" />
	<node pkg = "gscam" type = "gscam" name = "web_cam">
		  <param name = "camera_name" value = "usb_cam" />
	</node>
	-->
			<!-- Taken from GSCAM config -->
	 <!-- <remap from = "camera/image_raw" to = "camera/visible/image" /> -->

	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" >
		<param name="video_device" value="/dev/video0" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="pixel_format" value="yuyv" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="io_method" value="mmap"/>
	</node>

	<!-- Alternatively use rqt_view_image, which is what we'll do for this assignment -->
	<!--
	<node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
		<remap from="image" to="/usb_cam/image_raw"/>
		<param name="autosize" value="true" />
	</node>
	-->

	<node name="motion_node" pkg="team6_assignment_4" type="motion_node" output="screen"/>

	<node name="motion_mode_keyboard" pkg="team6_assignment_4" type="motion_mode_keyboard" output="screen"/>
</launch>
